scalyr_agent/monitor_utils/k8s.py

```diff
@@ -1451,21 +1490,26 @@ def query_api( self, path, pretty=0, return_temp_errors=False, rate_limited=Fals
 
         response = None
 
-        # Optionally save api json to disk
-        log_responses = self.log_api_responses_to_disk and self.agent_log_path
-        logged_response_file = None
-        if log_responses:
-            logged_response_file = self.__open_api_response_log(path, rate_limited)
-        try:
-            # Optionally prepend stack trace into logged response file
-            if logged_response_file:
-                traceback.print_stack(file=logged_response_file)
-                logged_response_file.write('\n\n')
+        # Various state used logging of responses
+        log_responses = self.log_api_responses
+        logged_response = []
+        response_status_code = -1
+        response_len = 0
 
-            if logged_response_file:
-                self.__check_for_fake_response(logged_response_file)
+        try:
+            # Optionally prepend stack trace into logged response
+            if log_responses:
+                limited_txt = ''
+                if rate_limited:
+                    limited_txt = ' (rate limited)'
+                logged_response.append('k8s.query_api%s: %s' % (limited_txt, path))
+                stack_trace = ''.join(traceback.format_stack())
+                logged_response.append('\\n%s' % stack_trace.replace('\n', '\\n'))
 
             # Make actual API call
+            latency = float('inf')
+            if log_responses and self.log_api_min_latency > 0:
```

Comment:
Can you make a new boolean `log_latency` or similar that equals the value of this condition, and then check against this variable in this if statement, and also in the following one on line 1523.  This will ensure that if the condition ever changes, we don't need to update multiple places.